packages/twenty-front/src/modules/object-record/record-merge/components/MergePreviewEffect.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  MergeManySettings,
+  useMergeManyRecords,
+} from '@/object-record/hooks/useMergeManyRecords';
+import { useUpsertRecordsInStore } from '@/object-record/record-store/hooks/useUpsertRecordsInStore';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { useEffect } from 'react';
+
+type MergePreviewEffectProps = {
+  objectNameSingular: string;
+  selectedRecords: ObjectRecord[];
+  mergeSettings: MergeManySettings;
+  onMergePreviewRecordChange: (record: ObjectRecord | null) => void;
```

Comment:
NIT but the names were a bit difficult to understand for me, I think I would have preferred something like:
```
  onPreviewChange: (record: ObjectRecord | null) => void;
  onLoadingChange: (isLoading: boolean) => void;
```